### PR TITLE
Team management page updates

### LINKED ISSUE #14608: Create Front End for adding or removing VLJs from Auto-Distribution of Cases
## User or job story
User story: As a DVC, I need the ability to add or remove VLJs from the case auto-distribution feature, so that judges do not receive cases while they are on leave. 

## Acceptance criteria
- [ ] Please put this work behind the feature toggle: Priority Case Auto-Distribution
- [ ] This feature should be accessible to the following user groups: DVCs
- [ ] Include screenshot(s) in the Github issue if there are front-end changes
- [ ] Update documentation: [link]
- [ ] DVC have the ability to add or remove judges from the case auto-distribution feature

## Release notes
DVCs ow have the ability to add or remove VLJs from receiving auto-distributed priority cases. 

### Technical notes
Auto-distribution will need to check if VLJ is allowed to receive auto-distributed cases. 

### Resources/other links
Product Brief: https://docs.google.com/document/d/1EjX-tYz0iI1hLP61p2p-O5rLvltj7NmlIofYRRwPMMs/edit#

### PR DESCRIPTION
**Pull request by hschallhorn**
Resolves #14608

### Description
Sneaky updates to the team management page as detailed in #14608 and discussed [in slack](https://dsva.slack.com/archives/CJL810329/p1597072079175800)

### Acceptance Criteria
- [x] No ID column
- [x] Order by team name
- [x] Use judge full names as team name
- [x] Use dvc full names as team name
- [x] Use button styling for org admin page
- [x] No URL column for non representatives (VSO and private bars)
- [x] No text field editing for non representatives (VSO and private bars)
- [x] No update button for non representatives (VSO and private bars)

### Testing Plan
1. Log in as TEAM_ADMIN and navigate to http://localhost:3000/team_management

### User Facing Changes

 BEFORE|AFTER
 ---|---
Screen Shot 2020-08-31 at 2 18 12 PM|Screen Shot 2020-08-31 at 2 17 47 PM

<details><summary>Part 2 #14928</summary>
### Description
Allows DVCs to access the team management page, but only view judge teams.

### Acceptance Criteria
- [ ] DVCs can navigate to the team management page from their main dropdown
- [ ] DVCs can view the team management page
- [ ] DVCs can only see judge teams on the team management page
- [ ] DVCs cannot navigate to judge team management page from the caseflow team management page (no org admin page link)

### Testing Plan
1. Run `make reset` or just the following commands
   ```ruby
   dvc = User.create(css_id: "BVATCOLLIER", station_id: 101, full_name: "Tonja DVCTeam Collier")
   dvc_team = DvcTeam.create_for_dvc(dvc)
   ```
1. Log in as BVATCOLLIER
1. Click "Caseflow team management" from your main dropdown

### User Facing Changes

 BVA ADMIN|DVC
 ---|---
Screen Shot 2020-08-31 at 2 32 45 PM|Screen Shot 2020-08-31 at 2 36 06 PM
</details>

<details><summary>Part 3 #14932</summary>
### Description
Adds toggles for DVCs to change whether or not a judge receives priority push cases. Shows the toggles to BVA admins, but does not allow them to change.

### Acceptance Criteria
- [ ] Toggles show for judge teams only
- [ ] DVC's can toggle said toggles
- [ ] BVA admins cannot but can still see said toggles
- [ ] Toggling said toggles changes the accepts_priority_pushed_cases field of the judge team

### Testing Plan
1. Run `make reset` or just the following commands
   ```ruby
   dvc = User.create(css_id: "BVATCOLLIER", station_id: 101, full_name: "Tonja DVCTeam Collier")
   dvc_team = DvcTeam.create_for_judge(judge)
   ```
1. Log in as BVATCOLLIER
1. Click "Caseflow team management" from your main dropdown
1. Play with the toggles while checking changes on the back end and changes persist through refreshing the page
```ruby
JudgeTeam.find_by(url: "bvaaabshire").accepts_priority_pushed_cases
```

### User Facing Changes

 BVA Admins|DVCs
 ---|---
Screen Shot 2020-08-31 at 2 42 03 PM|Screen Shot 2020-08-31 at 2 40 02 PM

### CODE REVIEW

File: app/controllers/team_management_controller.rb [hunk 1]
```diff
@@ -8,9 +8,9 @@ def index
       format.html { render template: "queue/index" }
       format.json do
         render json: {
-          judge_teams: JudgeTeam.all.order(:id).map { |jt| serialize_org(jt) },
-          private_bars: PrivateBar.all.order(:id).map { |private_bar| serialize_org(private_bar) },
-          vsos: Vso.all.order(:id).map { |vso| serialize_org(vso) },
+          judge_teams: JudgeTeam.order(:name).map { |jt| serialize_org(jt) },
+          private_bars: PrivateBar.order(:name).map { |private_bar| serialize_org(private_bar) },
+          vsos: Vso.order(:name).map { |vso| serialize_org(vso) },
```
Comment (hschallhorn) [inline app/controllers/team_management_controller.rb:13]: - [ ] Order by team name

`.all` is not needed here

File: app/controllers/team_management_controller.rb [hunk 2]
```diff
@@ -70,13 +70,13 @@ def update_params
   end
 
   def other_orgs
-    Organization.all.order(:id).reject { |org| org.is_a?(JudgeTeam) || org.is_a?(Representative) }
+    Organization.order(:name).reject { |org| org.is_a?(JudgeTeam) || org.is_a?(Representative) }
```
Comment (hschallhorn) [inline app/controllers/team_management_controller.rb:73]: - [ ] Order by team name

`.all` is not needed here

File: app/controllers/team_management_controller.rb [hunk 3]
```diff
@@ -70,13 +70,13 @@ def update_params
...
   end
 
   def serialize_org(org)
     {
       id: org.id,
-      name: org.name,
+      name: org.is_a?(JudgeTeam) ? org.judge.full_name.titleize : org.name,
```
Comment (hschallhorn) [inline app/controllers/team_management_controller.rb:79]: - [ ]  Use judge full names as team name

File: client/app/queue/TeamManagement.jsx [hunk 4]
```diff
@@ -153,27 +153,26 @@ class OrgList extends React.PureComponent {
...
 
 OrgList.defaultProps = {
-  showBgsParticipantId: false
+  isRepresentative: false
 };
 
 OrgList.propTypes = {
   orgs: PropTypes.array,
-  showBgsParticipantId: PropTypes.bool
+  isRepresentative: PropTypes.bool
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:175]: We show participant ids for all representatives. We also allow editing of names, urls, and participant ids for all representatives. Rename bool to be less specific

File: client/app/queue/TeamManagement.jsx [hunk 5]
```diff
@@ -222,14 +221,14 @@ class OrgRow extends React.PureComponent {
   // TODO: Indicate that changes have been made to the row by enabling the submit changes button. Default to disabled.
   render = () => {
     return <tr>
-      <td>{ this.props.id }</td>
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:225]: - [ ] No ID column

File: client/app/queue/TeamManagement.jsx [hunk 6]
```diff
@@ -153,27 +153,26 @@ class OrgList extends React.PureComponent {
   render = () => {
     return <React.Fragment>
       <tr {...labelRowStyling}>
-        <td>{COPY.TEAM_MANAGEMENT_ID_COLUMN_HEADING}</td>
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:156]: - [ ] No ID column

File: client/app/queue/TeamManagement.jsx [hunk 7]
```diff
@@ -222,14 +221,14 @@ class OrgRow extends React.PureComponent {
...
     return <tr>
-      <td>{ this.props.id }</td>
       <td>
         <TextField
           name={`${COPY.TEAM_MANAGEMENT_NAME_COLUMN_HEADING}-${this.props.id}`}
           label={false}
           useAriaLabel
           value={this.state.name}
           onChange={this.changeName}
+          readOnly={!this.props.isRepresentative}
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:231]: - [ ]  No text field editing for non representatives (VSO and private bars)

File: client/app/queue/TeamManagement.jsx [hunk 8]
```diff
@@ -239,10 +238,11 @@ class OrgRow extends React.PureComponent {
           useAriaLabel
           value={this.state.url}
           onChange={this.changeUrl}
+          readOnly={!this.props.isRepresentative}
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:241]: - [ ]  No text field editing for non representatives (VSO and private bars)

File: client/app/queue/TeamManagement.jsx [hunk 9]
```diff
@@ -253,29 +253,36 @@ class OrgRow extends React.PureComponent {
...
-          onClick={this.submitUpdate}
-        />
+        { this.props.isRepresentative &&
+          <Button
+            name={COPY.TEAM_MANAGEMENT_UPDATE_ROW_BUTTON}
+            id={`${this.props.id}`}
+            classNames={['usa-button-secondary']}
+            onClick={this.submitUpdate}
+          />
+        }
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:263]: - [ ] No update button for non representatives (VSO and private bars)

File: client/app/queue/TeamManagement.jsx [hunk 10]
```diff
@@ -253,29 +253,36 @@ class OrgRow extends React.PureComponent {
...
       </td>
       <td>
-        { this.state.url && <Link to={this.state.user_admin_path}>Org admin page</Link> }
+        { this.state.url && <Link to={this.state.user_admin_path}>
+          <Button
+            name="Org Admin Page"
+            classNames={['usa-button-secondary']}
+          />
+        </Link> }
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:271]: - [ ] Use button styling for org admin page

File: client/app/queue/TeamManagement.jsx [hunk 11]
```diff
@@ -222,27 +221,28 @@ class OrgRow extends React.PureComponent {
...
           label={false}
           useAriaLabel
           value={this.state.name}
           onChange={this.changeName}
+          readOnly={!this.props.isRepresentative}
         />
       </td>
       <td>
-        <TextField
+        { this.props.isRepresentative && <TextField
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:235]: - [ ] No URL column for non representatives (VSO and private bars)

File: client/app/queue/TeamManagement.jsx [hunk 12]
```diff
@@ -153,27 +153,26 @@ class OrgList extends React.PureComponent {
   render = () => {
     return <React.Fragment>
       <tr {...labelRowStyling}>
-        <td>{COPY.TEAM_MANAGEMENT_ID_COLUMN_HEADING}</td>
         <td>{COPY.TEAM_MANAGEMENT_NAME_COLUMN_HEADING}</td>
-        <td>{COPY.TEAM_MANAGEMENT_URL_COLUMN_HEADING}</td>
-        <td>{ this.props.showBgsParticipantId && COPY.TEAM_MANAGEMENT_PARTICIPANT_ID_COLUMN_HEADING}</td>
+        <td>{this.props.isRepresentative && COPY.TEAM_MANAGEMENT_URL_COLUMN_HEADING}</td>
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:157]: - [ ] No URL column for non representatives (VSO and private bars)

File: app/controllers/team_management_controller.rb [hunk 13]
```diff
@@ -70,17 +70,10 @@ def update_params
...
   def serialize_org(org)
-    {
-      id: org.id,
-      name: org.name,
-      participant_id: org.participant_id,
-      type: org.type,
-      url: org.url,
-      user_admin_path: org.user_admin_path
-    }
+    org.serialize
```
Comment (hschallhorn) [inline app/controllers/team_management_controller.rb:77]: Pull out to org

File: app/models/organizations/judge_team.rb [hunk 14]
```diff
@@ -64,6 +64,10 @@ def selectable_in_queue?
     false
   end
 
+  def serialize
+    super.merge(name: judge.full_name.titleize)
```
Comment (hschallhorn) [inline app/models/organizations/judge_team.rb:68]: - [ ]  Use judge full names as team name

File: app/models/organizations/dvc_team.rb [hunk 15]
```diff
@@ -31,4 +31,8 @@ def can_receive_task?(_task)
   def selectable_in_queue?
     false
   end
+
+  def serialize
+    super.merge(name: dvc.full_name.titleize)
```
Comment (hschallhorn) [inline app/models/organizations/dvc_team.rb:36]: - [ ] Use dvc full names as team name

File: app/controllers/team_management_controller.rb [hunk 16]
```diff
@@ -83,17 +83,10 @@ def update_params
...
   end
 
   def serialize_org(org)
-    {
-      id: org.id,
-      name: org.name,
-      participant_id: org.participant_id,
-      type: org.type,
-      url: org.url,
-      user_admin_path: org.user_admin_path
```
Comment (lomky) [inline app/controllers/team_management_controller.rb:96]: where did this end up going to?

Comment (lomky) [reply app/controllers/team_management_controller.rb:96]: works in the final PR - not sweating it!

**Review by lomky — APPROVED** [review]
:man_artist: :pray: :woman_technologist:

File: app/controllers/team_management_controller.rb (hunk 16 above)
Comment (hschallhorn) [reply app/controllers/team_management_controller.rb:96]: Fixed!

**Pull request merged**